js/nomNom.js

```diff
@@ -0,0 +1,197 @@
+					fourSquare.config.foodSpots[i] = {};
+					fourSquare.config.foodSpots[i].name = this.name;
+					fourSquare.config.foodSpots[i].address = this.location.formattedAddress[0];
+					fourSquare.getNumberOfLikes(this.id, i);
+					fourSquare.getImagePath(this.id, i);
+				});
+			},
+			error: function(json) {
+				console.log("Error pinging FourSquare");
+			}
+		}).done(function(){
+			setTimeout(function(){fourSquare.orderResultsByRank();},2000);
+		});	
+	},
+	getNumberOfLikes : function(resultsId,index) {
+		var httpRequest = 'https://api.foursquare.com/v2/venues/' + resultsId + '/likes?client_id=5TN04INYIF1T2C2QIO24I1YV2R41KTDN0YAYIRKRANEPUAPS&client_secret=0MO2N2SNHPNYF21LGYQSTYB3USPDXXEIJXV40UFLQDF4IHAJ&v=20130815';
+		$.ajax({
+			xhrFields: {
+				withCredentials: false
+			},
+			crossDomain: true,
+			type: "GET",
+			url: httpRequest,
+			dataType: "json",
+			success: function(json) {
+				fourSquare.config.foodSpots[index].likes = json.response.likes.count;
+			},
+			error: function(json) {
+				console.log("Error getting Likes");
+			}
+		});	
+	},
+	getImagePath : function(resultsId,index) {
+		var httpRequest = 'https://api.foursquare.com/v2/venues/' + resultsId +'/photos?client_id=5TN04INYIF1T2C2QIO24I1YV2R41KTDN0YAYIRKRANEPUAPS&client_secret=0MO2N2SNHPNYF21LGYQSTYB3USPDXXEIJXV40UFLQDF4IHAJ&v=20130815';
+		$.ajax({
+			xhrFields: {
+				withCredentials: false
+			},
+			crossDomain: true,
+			type: "GET",
+			url: httpRequest,
+			dataType: "json",
+			success: function(json) {
+				if (json.response.photos.count === 0)
+						fourSquare.config.foodSpots[index].img = '../img/noimage.png';
+				else
+					fourSquare.config.foodSpots[index].img = json.response.photos.items[0].prefix + '200x200' + json.response.photos.items[0].suffix;
+			},
+			error: function(json) {
+				console.log("Error getting Images");
+			}
+		});	
+	}
+};
+
+var showMore = {
+	init : function() {
+		$('#show-more').on('click',function(){
+			startCell = startCell + 10;
+			showMore.loadNewData();
+		});
+	},
+	loadNewData : function() {
+		populateTheCells.init(startCell);
+	}
+};
+
+
+var zipEntry = {
+	init : function() {
+		zipEntry.bindEnter();
+	},
+	bindEnter : function() {
+		$(document).keypress(function(e){
+			if (e.which === 13){
+				zipEntry.entered();
+			}
+		});
+	},
+	entered : function() {
+		zipCode = Number($('#zip-code').val());
+		$('#show-more').find('p').text("Loading...");
+		startCell = 0;
+		fourSquare.init();
+	}
+};
+
+
+
+//on document ready
+$(function() {
+	//build the html for the rows
+	$(rows).each(function(i){
+		rows[i].populateHtml();
+	});
+	
+	showMore.init();
+	zipEntry.init();
+
+	//Wait for the Zip Code before pinging Four Square
+	function checkForZip(){
+		if (zipCodeFlag && zipCode !== "undefined") {
+			console.log("checking for resturants in the " + zipCode + " area");
+			fourSquare.init();
+		} else {
+			setTimeout(function(){checkForZip();},1000);
+		}
+	}
```

Comment:
Rather than checking for `zipCodeFlag` and `zipCode` repeatedly, why not just attach to the geolocation callback? Then you know exactly when it comes back, or if it times out.